src/audio/eq_fir.c

```diff
@@ -187,7 +187,7 @@ static void eq_fir_s32_passthrough(struct fir_state_32x16 fir[],
 	int32_t *dest = (int32_t *)sink->w_ptr;
 	int n = frames * nch;
 
-	memcpy(dest, src, n * sizeof(int32_t));
+	memcpy_s(dest, n * sizeof(int32_t), src, n * sizeof(int32_t));
```

Comment:
This should rather check the distance between r_ptr/w_ptr and their respective end of buffers. Pls check other similar cases in eq_... Some single comp_buffer_copy(struct comp_buffer *source) that performs necessary checks similar to comp_buffer_can_copy_bytes() and calls memcpy_s() could be nice.